tools/ts_proto/bundle-proto.js

```diff
@@ -0,0 +1,46 @@
+import {pbjs, pbts} from 'protobufjs-cli';
+
+const [path, fileType] = process.argv.slice(2);
+
+if (fileType === 'js') {
+  process.exit(
+    await new Promise((resolve) => {
+      pbjs.main(['--target', 'static-module', '-w', 'es6', path], function (err, output) {
+        if (err) {
+          process.stderr.write(err.name);
+          process.stderr.write(err.message);
+          process.stderr.write(err.stack);
+          return resolve(1);
+          
+        }
+
+        output = output.replace(
+          'import * as $protobuf from "protobufjs/minimal";',
+          'import $protobuf from "protobufjs";',
```

Comment:
Doesn't ESBuild handle this properly? Has this come up recently — or did it never work and we just didn't notice?

Reply:
It seems brittle to do it the current way with a custom bundler. There are alternative Protobuf generation tools, or we could just have a temporary Bazel rule in between like `expand_template` to substitute things.

Just want to avoid having to clean this up in the future – if we can just move to something better. Also we should link to an issue so that ProtobufJS-cli can support ESM properly.

Reply:
I looked into the code more and found that we can do it another way.  PTAL